### DISCUSSION TITLE
Is there a method on_model_change or is it planned?

### DISCUSSION THREAD

**Discussion by dima23113** [aminalaee/sqladmin#341] [Ideas]
Hi!  Is there a method on_model_change or is it planned? there is such a method in the flask admin and it allows you to intercept the data before saving. It turns out an analogue of the save method in Django. or maybe there is some solution? I need to fill in the field before saving to the database, I did not find a solution.

`
class PostView(ModelView):

    def after_model_change(self, form, model, is_created):
        print(form, model, is_created)

    def on_model_change(self, form, model, is_created):
        print(form, model, is_created)
`

**Comment by aminalaee** [comment]
Hey, thanks for the suggestion.
Right now there is no such method, but we can add it with the same API as flask-admin. I think it's pretty simple to add. We just add an empty function signature in sqladmin that is called before and after update so users can override it.

Feel free to create a PR otherwise I'll do it whenever I can.

**Reply by dima23113** [reply]
Thanks and have a nice day! :)
